src/devices/Ads1115/Ads1115.cs

```diff
@@ -623,50 +592,37 @@ public ElectricPotential MaxVoltageFromMeasuringRange(MeasuringRange measuringRa
         /// </summary>
         /// <param name="dataRate">One of the <see cref="DataRate"/> enumeration members.</param>
         /// <returns>A frequency, in Hertz</returns>
-        public double FrequencyFromDataRate(DataRate dataRate)
+        public double FrequencyFromDataRate(DataRate dataRate) => dataRate switch
         {
-            switch (dataRate)
-            {
-                case DataRate.SPS008:
-                    return 8.0;
-                case DataRate.SPS016:
-                    return 16.0;
-                case DataRate.SPS032:
-                    return 32.0;
-                case DataRate.SPS064:
-                    return 64.0;
-                case DataRate.SPS128:
-                    return 128.0;
-                case DataRate.SPS250:
-                    return 250.0;
-                case DataRate.SPS475:
-                    return 475.0;
-                case DataRate.SPS860:
-                    return 860.0;
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(dataRate), "Unknown data rate used");
-            }
-        }
+                DataRate.SPS008 => 8.0,
+                DataRate.SPS016 => 16.0,
+                DataRate.SPS032 => 32.0,
+                DataRate.SPS064 => 64.0,
+                DataRate.SPS128 => 128.0,
+                DataRate.SPS250 => 250.0,
+                DataRate.SPS475 => 475.0,
+                DataRate.SPS860 => 860.0,
+                _ => throw new ArgumentOutOfRangeException(nameof(dataRate), "Unknown data rate used")
+        };
 
         /// <summary>
         /// Cleanup.
         /// Failing to dispose this class, especially when callbacks are active, may lead to undefined behavior.
         /// </summary>
         public void Dispose()
         {
-            if (_i2cDevice != null)
+            if (_i2cDevice is object)
             {
                 DisableAlertReadyPin();
-                _i2cDevice?.Dispose();
+                _i2cDevice.Dispose();
                 _i2cDevice = null!;
             }
 
-            if (_shouldDispose && _gpioController != null)
+            if (_shouldDispose)
             {
-                _gpioController.Dispose();
+                _gpioController?.Dispose();
+                _gpioController = null;
```

Comment:
Moving the nulling of `_gpioController` here means that if we called the constructor like
`new Ads1115(i2cDevice, gpioController, 1, false, ...)` then we would still keep the reference to the pass-in `gpioController` unlike the old code. Is this an intended change?

Reply:
You are right.  It would be better the old way. I was conflating the call to dispose and the assignment to `null` as related operations. They are not. Thanks for catching this.

Filed an issue on that: https://github.com/dotnet/iot/issues/1272